data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -80,6 +78,18 @@ then
     record_service_stage_success
 fi
 
+ICSP_MANIFEST_FILE="$PWD/manifests/image-content-source-policy-0.yaml"
+IDMS_MANIFEST_FILE="$PWD/manifests/image-digest-mirror-set-0.yaml"
```

Comment:
Isn't this making the assumption that the first entry in the image digest source list is the one that provides the mirror for the release image?
We previously had that assumption for the agent-based installer, but I believe we dropped it before 4.12 shipped. And IPI has never relied that assumption.

The installer knows what version it is for, maybe we can just template that straight in here instead of calling `oc` at all?

Reply:
Alternatively if we pass `<(cat "$PWD/manifests/image-digest-mirror-set-*.yaml")` or something, maybe that will work.

Reply:
This is passing in the entire icsp/idms file in the param so its not relying on the first entry, its similar to what we do with assisted-service in https://github.com/openshift/assisted-service/pull/4684

Reply:
The installer appears to split each entry into a separate file, just to be difficult :frowning_face: 
https://github.com/openshift/installer/blob/master/pkg/asset/manifests/imagedigestmirrorset.go#L45-L77

Reply:
Oh. I see what you mean. Yeah we need the entire file.

Reply:
Although this PR may work (and appears to) its not a complete solution as all of the mirrored sets aren't included in the `-0.yaml` file. As a result the order if the entries in install-config.yaml could determine whether or not the the version is picked up.

Let's continue to merge this while I work on a more complete solution.

Reply:
I'm not sure the installer does know what version it is... at least when it's not being built by ART. If we do know the version,  +1 to just templating that in.

> The installer appears to split each entry into a separate file, just to be difficult frowning_face https://github.com/openshift/installer/blob/master/pkg/asset/manifests/imagedigestmirrorset.go#L45-L77 https://github.com/openshift/installer/blob/master/pkg/asset/manifests/imagesourcepolicy.go#L46-L73

I [asked](https://redhat-internal.slack.com/archives/C68TNFWA2/p1683826228773909?thread_ts=1683820476.478359&cid=C68TNFWA2) @QiWang19 about this briefly. Based on this conversation and that one,  I'm not sure that what we have is what we want. Should we just have one IMDS file with all of the mirrors contained in the [`ImageDigestMirrors`](https://redhat-internal.slack.com/archives/C68TNFWA2/p1683826228773909?thread_ts=1683820476.478359&cid=C68TNFWA2)? That makes the most sense to me.

Reply:
Having one file for all the ICSP (or IDMS) entries would work. Alternatively we can add a Version field to the bootstrapTemplateData and retrieve it by using the `oc` command with an icsp file built using the Registries info https://github.com/openshift/installer/blob/master/pkg/asset/ignition/bootstrap/common.go#L75

Reply:
I changed to use one file for both ICSP and IDMS and include all the mirror entries in it. PTAL.

Reply:
@patrickdillon what's our policy on changing the generated manifests like this?
Note that this change doesn't only affect the layout of the manifest files, but also what gets installed in the cluster (1 ICSP with multiple entries, instead of multiple ICSPs with one entry each). This makes me wonder why we were doing it the other way in the first place.

Reply:
Yeah when I changed the generated name to remove the `-0` suffix it looks like it caused a conflict with a manifest in the cluster - see the error May posted. So I changed it to use the same filename as before e.g. `image-content-source-policy-0.yaml` but now it contains all the entries. Its a good question as to how that will affect the resulting cluster.

We could go to the alternative plan I suggested above which is to build a tmp icsp/idms file in bootstrap/common.go and use `oc` to get the version and template it.

Reply:
> This makes me wonder why we were doing it the other way in the first place.

+1.  I will ping @bparees & @QiWang19 to see if they know of a good reason for splitting the registries to multiple files.

> what's our policy on changing the generated manifests like this?

Manifests are not regarded as a full API, so I think it's mostly Hyrum's Law for this. But IMDS has not been released yet, so I think that's a very safe change.